src/oc/auth/api/slack.clj

```diff
@@ -248,23 +257,19 @@
                                                 (assoc :admin (user-res/admin-of conn (:user-id user)))
                                                 (assoc :slack-id (:slack-id slack-user))
                                                 (assoc :slack-token (:slack-token slack-user))) :slack)
-
             ;; Determine where we redirect them to
             bot-only? (and target-team ((set (:slack-orgs target-team)) (:slack-org-id slack-org)))
-            redirect-arg (if bot-only? :bot :team)]
+            redirect-arg (if (and (not bot-only?) slack-org (not (contains? slack-org :bot-token))) :bot :team)]
```

Comment:
@thepug why `bot-only?` check has become `(not bot-only?)`? I'm didn't check the rest of the logic of this but i think the old `bot-only?` should still be working.
In fact if you signup the first time, skip the bot, then go to settings and add the bot you are redirected there with `access=team` not bot.

Reply:
Yeah, I was trying to keep the same variable names but maybe that isn't a good idea.  The `bot-only?` wasn't working as I expected.  I'll give it another round with the add to team test.